cluster-autoscaler/expander/grpcplugin/grpc_client_test.go

```diff
@@ -263,13 +286,15 @@ func TestBestOptionsErrors(t *testing.T) {
 	}
 	for _, tc := range testCases {
 		grpcNodeInfoMap := populateNodeInfoForGRPC(tc.nodeInfo)
-		mockClient.EXPECT().BestOptions(
-			gomock.Any(), gomock.Eq(
-				&protos.BestOptionsRequest{
-					Options: []*protos.Option{&grpcEoT2Micro, &grpcEoT2Large, &grpcEoT3Large, &grpcEoM44XLarge},
-					NodeMap: grpcNodeInfoMap,
-				})).Return(&tc.mockResponse, tc.errResponse)
-		resp := g.BestOptions(options, tc.nodeInfo)
+		if tc.client.grpcClient != nil {
```

Comment:
Why is this needed? It seems like if this happen, the test case haven't been properly initialized, in which case it should fail and not pass silently

Reply:
@MaciekPytel  it was for this test case https://github.com/kubernetes/autoscaler/blob/29de5009060c12ef95d70eaacfdc6bc83c151524/cluster-autoscaler/expander/grpcplugin/grpc_client_test.go#L259-L263, prior to my change, the test was not executing the correct code path since `tc.client` was never being used, it was always `g` being used as the client https://github.com/kubernetes/autoscaler/pull/7351/files#diff-5392da3174a0a9693626803662fd3c6871211cfe739ec3b7d6a5efff38e1cae9L272